go/vt/vttablet/tabletserver/connpool/dbconn.go

```diff
@@ -329,15 +328,30 @@ func (dbc *DBConn) ApplySettings(ctx context.Context, settings []string) error {
 		if _, err := digest.WriteString(q); err != nil {
 			return err
 		}
-		s.WriteString(q)
 	}
-	dbc.settings = s.String()
+	dbc.settings = settings
 	dbc.settingsHash = digest.Sum64()
 	return nil
 }
 
-func (dbc *DBConn) SettingHash() uint64 {
-	return dbc.settingsHash
+func (dbc *DBConn) IsSettingsApplied() bool {
+	return len(dbc.settings) > 0
+}
+
+func (dbc *DBConn) IsSameSetting(settings []string) bool {
+	return compareStringSlice(settings, dbc.settings)
```

Comment:
`slices.Equal(settings, dbc.settings)` should be enough

Reply:
I could not find such library :(

Reply:
ah, it's an experimental package. Didn't realise it.
`golang.org/x/exp/slices`